doc/source/whatsnew/v1.4.2.rst

```diff
@@ -23,6 +23,7 @@ Fixed regressions
 
 Bug fixes
 ~~~~~~~~~
+- Fixed metadata propagation in :meth:`NDFrame.xs`, :meth:`DataFrame.iterrows`, :meth:`DataFrame.loc` and :meth:`DataFrame.iloc` (:issue:`28283`)
```

Comment:
Yes it is*, and sorry for the delayed response. I wanted to validate this again, more extensively. I've made some slight changes to the code, for testing purposes only, so that every single call to loc and iloc adds attrs to the original object and validates that they still exist on the returned object. After that, I ran the entire test suite with these additional checks built in. Assuming that this gives 100% coverage of all branches of loc and iloc, there were only three failures, all of which were related to _get_item_cache.
Basically, if attrs/_metadata/flags change after an object was cached, the next time a sliced object is returned using _get_item_cache, it will have the old attrs/_metadata/flags. I'm not sure if we'd want to add a finalize call to _get_item_cache though? If attrs/_metadata/flags are considered constants for each object, then this might not be a problem. This only came up because my extended testing modified attrs between multiple loc/iloc calls, but was never caused by a test itself.